lib/lm_docstache/conditional_block.rb

```diff
@@ -0,0 +1,88 @@
+require 'strscan'
+
+module LMDocstache
+  class ConditionalBlock
+    BLOCK_MATCHER = LMDocstache::Parser::BLOCK_MATCHER
+
+    attr_reader :elements, :condition, :value
+
+    def initialize(elements:, condition:, content: nil)
+      @elements = elements
+      @condition = condition
+      @content = content
+      @evaluated = false
+    end
+
+    def content
+      return @content if inline?
+    end
+
+    def evaluate_with_value!(value)
+      return false if evaluated?
+
+      inline? ? evaluate_inline_block!(value) : evaluate_multiple_nodes_block!(value)
+
+      @evaluated = true
+    end
+
+    def evaluated?
+      !!@evaluated
+    end
+
+    def inline?
+      @elements.size == 1
+    end
+
+    def self.inline_blocks_from_paragraph(paragraph)
+      node_set = Nokogiri::XML::NodeSet.new(paragraph.document, [paragraph])
+      conditional_blocks = []
+      scanner = StringScanner.new(paragraph.text)
+      matches = []
+
+      while scanner.scan_until(BLOCK_MATCHER)
+        next if matches.include?(scanner.matched)
+
+        content = scanner.captures[4]
```

Comment:
not a big deal, but would be nice to have a comment with an example of the `scanner.captures` content :)

Reply:
I agress with @daniloisr that's not very clear what the results are.

Reply:
Done in 640e537.